kobo/apps/stripe/tests/test_organization_usage.py

```diff
@@ -310,6 +312,34 @@ def test_plan_canceled_last_month(self):
             == current_billing_period_end.isoformat()
         )
 
+    def test_plan_canceled_edge_date(self):
+        """
+        If a plan is canceled on the last day of a 31-day month, we want the subsequent
+        billing cycle to end on the last day of the next month, but we also need to make
+        sure the cycle starts on the cancelation date
+        """
+        cancel_date = datetime(year=2024, month=8, day=31, tzinfo=ZoneInfo('UTC'))
+        with freeze_time(cancel_date.replace(day=1)):
+            subscription = generate_plan_subscription(self.organization)
```

Comment:
I would use a separate variable (or maybe only a comment?) to make the purpose of `freeze_time` obvious.

```
        cancel_date = datetime(year=2024, month=8, day=31, tzinfo=ZoneInfo('UTC'))
        frozen_datetime_now = cancel_date.replace(day=1)
        with freeze_time(frozen_datetime_now):
            subscription = generate_plan_subscription(self.organization)
```

What do you think?

Reply:
Yeah, I think that's a good idea. I was trying to be economical with line length, but it makes more sense to make the code explain itself.